### PR TITLE
#16627 convert weights using torch.as_tensor to avoid warning

### LINKED ISSUE #16627: WeightedRandomSampler throws warning for copy constructor
## 🐛 Minor Bug

Passing a `weights` tensor to `WeigthedRandomSampler` throws a warning:
> UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).

This is due to [this line of code](https://github.com/pytorch/pytorch/blob/279238f0b81dce43636ebc8c7b1c45cc98c7a60d/torch/utils/data/sampler.py#L119).

Reported by Wafaa_Wardah in [this post](https://discuss.pytorch.org/t/some-problems-with-weightedrandomsampler/23242/11?u=ptrblck).

## To Reproduce

```python
import torch
from torch.utils.data import WeightedRandomSampler

weights = torch.randn(10)
sampler = WeightedRandomSampler(weights=weights, num_samples=10)
```

## Environment

 - PyTorch Version (e.g., 1.0): 1.0.0.dev20190104

## Possible bug fix:

I think the easiest way to fix this issue would be to change the current line from
```python
self.weights = torch.tensor(weights, dtype=torch.double)
```
to
```
self.weights = torch.as_tensor(weights, dtype=torch.double)
```

CC @SsnL Let me know, if that would be sufficient and I can create a fast PR for this.

Best,
ptrblck

### PR DESCRIPTION
**Pull request by ptrblck**
Minor change which fixes #16627

### CODE REVIEW

**Review by soumith — APPROVED** [review]
ci failures are unrelated
